app/api/templates/reindex.js

```diff
@@ -22,6 +22,11 @@ const SHOULD_NOT_TRIGGER_REINDEX = [
 
 function compareTemplateProperties(updatedProperties, originalProperties) {
   const changedProps = [];
+
+  if (updatedProperties.length !== originalProperties.length) {
+    changedProps.push('PROPERTY_ADDED');
+  }
+
```

Comment:
@grafitto I think the length of the properties should not be the criteria for check for added properties because I can delete one property and add a new one and the length will be the same. Check the _ids of the properties instead. One case where it's important to reindex after adding a property is for those that have `generatedid` type because for these properties it will generate the values in the entities.

Reply:
Working on this